### PR TITLE
Fix data races and change storage format to HTML

### PR DESCRIPTION
**Pull request by vladikoff**
@Natim this fixes a few bugs:

Fixes https://github.com/mozilla/notes/issues/421
Fixes https://github.com/mozilla/notes/issues/424
Fixes https://github.com/mozilla/notes/issues/420
Fixes https://github.com/mozilla/notes/issues/406
Fixes https://github.com/mozilla/notes/issues/407
Fixes https://github.com/mozilla/notes/issues/428

Related read for the reason to switch to HTML formatting: https://github.com/ckeditor/ckeditor5-markdown-gfm/issues/4

### CODE REVIEW

**Comment by vladikoff** [comment]
@Natim here's the only change I made in the gfm processor: https://github.com/vladikoff/ckeditor5-markdown-gfm/commit/407840c309235750370868c511e6390a50621e49

I feel like storing the data as HTML is the way to go, especially if we plan to have COLOR and RTL support in the future.

**Comment by vladikoff** [comment]
@Natim I feel like we should include this in the release because the multi window data race is pretty bad

**Comment by vladikoff** [comment]
Ah changing the Heading does not fire `text-change`

**Comment by vladikoff** [comment]
@Natim Ok should be good to go 👍 , test it out

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "@ckeditor/ckeditor5-essentials": "^1.0.0-alpha.2",
     "@ckeditor/ckeditor5-heading": "vladikoff/ckeditor5-heading#short-heading",
     "@ckeditor/ckeditor5-list": "^1.0.0-alpha.2",
-    "@ckeditor/ckeditor5-markdown-gfm": "^1.0.0-alpha.2",
+    "@ckeditor/ckeditor5-markdown-gfm": "vladikoff/ckeditor5-markdown-gfm#data-as-html",
```
Comment (Natim) [inline package.json]: Can we make this part of this repository?

Comment (vladikoff) [reply package.json]: i will try!!

Comment (vladikoff) [reply package.json]: Fixed by a function override instead of using the fork

Comment (Natim) [reply package.json]: So shall we get back to `^1.0.0-alpha.2`?

**Pull request merged**